### PR TITLE
[REF][PHP8.1] Fix some further places where passing in NULL is deprec…

### PR DESCRIPTION
**Pull request by seamuslee001**
…ated in php8.1

ping @eileenmcnaughton @demeritcowboy @totten @colemanw

### CODE REVIEW

File: IDS/Converter.php [hunk 1]
```diff
@@ -158,7 +158,7 @@ public static function convertFromJSCharcode($value)
                 if (preg_match_all('/\d*[+-\/\* ]\d+/', $char, $matches)) {
                     $match = preg_split('/(\W?\d+)/',
                                         (implode('', $matches[0])),
-                                        null,
+                                        '',
```
Comment (demeritcowboy) [inline IDS/Converter.php:161]: 3rd param is an int for preg_split?

Comment (demeritcowboy) [reply IDS/Converter.php:161]: Otherwise PR looks good.

Comment (seamuslee001) [reply IDS/Converter.php:161]: ah good point will update to -1

Comment (seamuslee001) [reply IDS/Converter.php]: ok fixed now @demeritcowboy

Comment (demeritcowboy) [reply IDS/Converter.php]: Cool. It turns out -1 and 0 mean the same thing in that spot (previously null would have been cast to 0).

**Comment by demeritcowboy** [comment]
Jenkins retest this please.

(Fails have been seen before I think as intermittent time-related, but let's check.)

**Pull request merged**